src/common/CudaDataUtils.hpp

```diff
@@ -42,6 +42,29 @@ __device__ __forceinline__ unsigned long long device_timer()
   return global_timer;
 }
 
+/*!
+ * \brief Method for launching a CUDA kernel with given configuration.
+ *
+ *        Note: method includes a call to check whether number of args
+ *              in kernel signature matches number of args passed to this 
+ *              method.
+ */
+template <typename... Args, typename F = void (*)(Args...)>
+void RPlaunchCudaKernel(F kernel,
+                        const dim3& numBlocks, const dim3& dimBlocks,
+                        std::uint32_t sharedMemBytes, cudaStream_t stream,
+                        Args const&... args) 
+{
+  constexpr size_t count = sizeof...(Args);
+  checkArgsCount(kernel, args...);
```

Comment:
Do we even need to check args count? Will the type system not link both Args and F to be consistent?